pipeline/extract.py

```diff
@@ -0,0 +1,115 @@
+"""
+Extraction script to find all sales tracking data we need from subscribed URL pages.
+
+1. Connect to RDS
+2. Query all URLs from the RDS
+3. Scrape the URL for: product_name,original_price,discount_price
+
+psql -h $DB_HOST -U $DB_USER -d $DB_NAME -p $DB_PORT -> to connect to db for testing
+
+
+"""
+import logging
+
+from connect_to_database import get_connection, get_cursor
+
+from dotenv import load_dotenv
+import requests
+from bs4 import BeautifulSoup
+
+QUERY_TO_FIND_URLS = """
+SELECT product_id,url FROM product;
+"""
+
+
+def extract_urls_from_db() -> list[list[int, str]]:
+    """Function to get urls from a database
+    returns the id of the product and its URL page in a list: [id,url]"""
+    conn = get_connection()
+
+    with get_cursor(conn) as db_cursor:
+        db_cursor.execute(QUERY_TO_FIND_URLS)
+        url_list = db_cursor.fetchall()
+
+    return url_list
+
+
+def get_html_from_url(web_page: str) -> bytes:
+    """ Gets the html content from a given URL"""
+    try:
+        html = requests.get(web_page, timeout=20)
+    except requests.exceptions.MissingSchema:
+        return "That URL does not exist."
+    except requests.exceptions.ConnectionError:
+        return "That URL does not exist."
```

Comment:
You could write a more specific error message for this `ConnectionError` as this error is more to do with failing to connect to the URL rather than the URL not existing, but otherwise, it's great that you're adding in error handling for this!

Reply:
Added this change